src/site/xdoc/cmdline.xml.vm

```diff
@@ -961,8 +988,77 @@ class Test {
   </code></pre>
 </div>
 
+<h4 id="Generating_Suppressions_XML"><span>
+7. Generating Suppressions XML (<span class="no-transform">-G, --generate-suppression</span>)
+</span></h4>
+<p><strong>Goal:</strong> Generate an XML file containing suppressions for all
+violations using the experimental suppression filter.</p>
+
+<span class ="inline-command" ><strong>Command:</strong> <code class="prettyprint">
+java -jar checkstyle-${projectVersion}-all.jar -c config.xml -G -o suppressions.xml Test.java
+</code></span>
+
+<p><strong>Content of config.xml:</strong></p>
+<div class="wrapper">
+  <pre class="prettyprint"><code class="language-xml">
+&lt;?xml version="1.0"?&gt;
+&lt;!DOCTYPE module PUBLIC
+  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+  "https://checkstyle.org/dtds/configuration_1_3.dtd"&gt;
+&lt;module name="Checker"&gt;
+  &lt;module name="TreeWalker"&gt;
+    &lt;module name="FallThrough"/&gt;
+  &lt;/module&gt;
+&lt;/module&gt;
+  </code></pre>
+</div>
+
+<p><strong>Content of Test.java:</strong></p>
+<div class="wrapper">
+  <pre class="prettyprint"><code class="language-java">
+/**
+ * Javadoc summary.
+ *
+ * Some description.
+ */
+class Test {
+  /**
+   * Some summary on method.
+   */
+  public void foo() {
+    int i = 0;
+    while (i >= 0) {
+      switch (i) {
+      case 1:
+      case 2:
+        i++;
+      case 3: // violation
+        i++;
+      }
+    }
+  }
+}
+  </code></pre>
+</div>
+
+<p><strong>Content of suppressions.xml:</strong></p>
+<div class="wrapper">
+  <pre class="prettyprint"><code class="language-xml">
+&lt;?xml version="1.0" encoding="UTF-8"?&gt;
+&lt;!DOCTYPE suppressions PUBLIC
+    "-//Checkstyle//DTD Checkstyle Configuration 1.2//EN"
+    "https://checkstyle.org/dtds/configuration_1_3.dtd"&gt;
+&lt;suppressions&gt;
+&lt;suppress
+       files="Test.java"
+       checks="FallThroughCheck"
```

Comment:
please add one more violation for different Check to target file to show that it will generate two suppression elements (with different Check) for same target filename.
Please give "id" for such new Check , to let user see that Check name or id will be used. If "id" defined, it will be used.